src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -3444,18 +3439,19 @@ class ScriptReferenceSegment(ansi.ObjectReferenceSegment):
 
 class ScriptContentSegment(BaseSegment):
     """This represents the script content.
-
     Because the script content could be written in
     LUA, PYTHON, JAVA or R there is no further verification.
     """
 
     type = "script_content"
-    match_grammar = Anything()
+    match_grammar = GreedyUntil(
+        Ref("FunctionScriptTerminatorSegment"),
+        enforce_whitespace_preceding_terminator=False,
+    )
```

Comment:
Limiting the `Anything()` grammar here by making it a `GreedyUntil()` terminated with the script terminator ensures that we don't overmatch.